homeassistant/components/energy/sensor.py

```diff
@@ -416,3 +416,18 @@ def update_config(self, config: dict) -> None:
     def native_unit_of_measurement(self) -> str | None:
         """Return the units of measurement."""
         return self.hass.config.currency
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the unique ID of the sensor."""
+        entity_registry = er.async_get(self.hass)
+        if (
+            registry_entry := entity_registry.async_get(
+                self._config[self._adapter.entity_energy_key]
+            )
+        ) and registry_entry.unique_id:
```

Comment:
It's not possible for a registry entry to not have a unique_id I think.
Is this to make the type checker happy? If it is, maybe something is wrong in the type definition of registry entries.

Reply:
You are right, its not needed :+1: